tests/middleware/test_gzip_middleware_unit.py

```
@@ -0,0 +1,554 @@
+"""
+Unit tests for GzipRequestMiddleware.
+
+These tests directly mock ASGI components to test middleware logic in isolation,
+complementing the integration tests in test_upload_endpoint_pvc.py.
+"""
+
+import gzip
+from unittest.mock import AsyncMock, Mock
```

Flake8 (7.3.0)

[error] 'unittest.mock.Mock' imported but unused

(F401)